setup/Core/HookGenTask.cs

```diff
@@ -1,90 +1,94 @@
 
-			if (!HookGen(tmlAssemblyPath, outputPath)) {
-				taskInterface.SetStatus("Cancelled");
-				return;
-			}
+			HookGen(TmlAssemblyPath, outputPath);
 
 			File.Delete(Path.ChangeExtension(outputPath, "pdb"));
 
-			MessageBox.Show("Success. Make sure you diff tModLoader after this");
+			success = true;
 		}
 
-		public static bool HookGen(string inputPath, string outputPath)
+		public override void FinishedPrompt()
 		{
-			string dotnetReferencesDirectory = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles) + installedNetRefs;
+			if (!success) {
+				return;
+			}
+
+			userPrompt.Inform("Success", "Success. Make sure you diff tModLoader after this");
+		}
+
+		private void HookGen(string inputPath, string outputPath)
+		{
+			string dotnetReferencesDirectory =
+				Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles) + InstalledNetRefs;
 
 			// Ensure that refs are present, for gods sake!
-			if (!Directory.Exists(dotnetReferencesDirectory) || Directory.GetFiles(dotnetReferencesDirectory, "*.dll").Length == 0) {
-				// Replace with exceptions if this is ever called in CLI.
-				MessageBox.Show(
-					$@"Unable to find reference libraries for .NET SDK '{dotnetSdkVersion}' - ""{dotnetReferencesDirectory}"" does not exist.",
-					$".NET SDK {dotnetSdkVersion} not found",
-					MessageBoxButton.OK
-				);
-
-				return false;
+			if (!Directory.Exists(dotnetReferencesDirectory) ||
+			    Directory.GetFiles(dotnetReferencesDirectory, "*.dll").Length == 0) {
+				throw new DirectoryNotFoundException($@"Unable to find reference libraries for .NET SDK '{DotnetSdkVersion}' - ""{dotnetReferencesDirectory}"" does not exist.");
 			}
 
 			using var mm = new MonoModder {
 				InputPath = inputPath,
 				OutputPath = outputPath,
 				ReadingMode = ReadingMode.Deferred,
-
 				DependencyDirs = { dotnetReferencesDirectory },
 				MissingDependencyThrow = false,
 			};
 
-			mm.DependencyDirs.AddRange(Directory.GetDirectories(binLibsPath, "*", SearchOption.AllDirectories));
+			mm.DependencyDirs.AddRange(Directory.GetDirectories(BinLibsPath, "*", SearchOption.AllDirectories));
 
 			mm.Read();
 
-			var gen = new HookGenerator(mm, "TerrariaHooks") {
-				HookPrivate = true,
-			};
+			var gen = new HookGenerator(mm, "TerrariaHooks") { HookPrivate = true };
 
-			foreach (var type in mm.Module.Types) {
-				if (!type.FullName.StartsWith("Terraria") || type.FullName.StartsWith("Terraria.ModLoader"))
+			foreach (TypeDefinition type in mm.Module.Types) {
+				if (!type.FullName.StartsWith("Terraria") || type.FullName.StartsWith("Terraria.ModLoader")) {
 					continue;
+				}
 
-				gen.GenerateFor(type, out var hookType, out var hookILType);
-				if (hookType == null || hookILType == null || hookType.IsNested)
+				gen.GenerateFor(type, out TypeDefinition hookType, out TypeDefinition hookILType);
+				if (hookType == null || hookILType == null || hookType.IsNested) {
 					continue;
```

Comment:
Probably just an overzealous autoformatter, but we do allow single line statements in our code style, and prefer `var`. Update the `.editorconfig` if you need to, but otherwise, try and avoid reformatting code that you aren't changing the semantics of as it makes review and `git blame` harder.

I made some diff reduction commits to help me review earlier

We can do a reformat commit after this PR is merged, with file scoped namespaces and proper brace style and that jazz. Refactor commits are easy to skip when blaming when listed as such.

Reply:
Rider doesn't seem to care about the severity of the code styles in the `.editorconfig`. `var style` and braces are configured as `silent` or `suggestion` but where still applied. I might have to change a setting somewhere.